### PR TITLE
Fix sending song ratings to last.fm.

### PR DESCRIPTION
**Pull request by JasonLG1979**
Now the sending of ratings (Love or Ban) is done when the rating happens via the rating-changed signal. Otherwise if the user did not actively rate the song during playback the rating will be reported at the end of the song.

### CODE REVIEW

**Comment by TingPing** [comment]
This was removed on purpose many years ago. The issue was that the meanings of love/ban on pandora do not directly match the rating system on last.fm. Now that we have a system in place for plugins having preferences I would not be against adding this as an option though.

**Comment by JasonLG1979** [comment]
They both have Love and Ban. What does Ban mean in Last.fm?

**Comment by JasonLG1979** [comment]
Love and Ban seem to pretty much mean the same thing in Pithos/Pandora and Last.fm.
http://www.last.fm/forum/34905/_/369421

**Comment by TingPing** [comment]
This decision was well before my time, I'll open up the launchpad issues later to find the discussion.

**Comment by JasonLG1979** [comment]
A ui option in the last.fm plugin seems to be the best idea I guess?

**Comment by TingPing** [comment]
https://bugs.launchpad.net/pithos/+bug/636600

tl;dr is it spams your last.fm account with "Favorited tracks" when all the user wanted to do was a pandora station to play more similar songs.

> A ui option in the last.fm plugin seems to be the best idea I guess?

An option should be fine.

**Comment by JasonLG1979** [comment]
Well I guess out this on hold until We/I can put a ui option in there and figure out how to check if a song has already been loved in the past (as not to spam)

**Comment by JasonLG1979** [comment]
*hold up on this
